src/native/libs/System.Native/pal_networkchange.c

```diff
@@ -175,8 +178,6 @@ Error SystemNative_ReadEvents(intptr_t sock, NetworkChangeEvent onNetworkChange)
 {
     (void)sock;
     (void)onNetworkChange;
-    // unreachable
-    abort();
-    return Error_SUCCESS;
+    return Error_EINVAL;
```

Comment:
Should we keep the old aborting code as a fallback and add separate #if for HAIKU instead? I am not familiar enough with the contract of this function though.

Reply:
I've added back the `abort()` logic, as an assertion that the function is unreachable.

This is true because on unsupported platforms, the higher-level C# libraries should already throw while trying to construct a network events socket.

Returning `Error_SUCCESS` does not make much sense though.